src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -0,0 +1,42 @@
+from datetime import date
+from typing import Optional
+
+from django.utils.translation import ugettext_lazy as _
+
+from openforms.submissions.models import Submission
+
+from ...base import BasePlugin
+from ...registry import register
+from .config import ObjectsAPIOptionsSerializer
+from .models import ObjectsAPIConfig
+
+
+@register("objects_api")
+class ObjectsAPIRegistration(BasePlugin):
+    verbose_name = _("Objects API registration")
+    configuration_options = ObjectsAPIOptionsSerializer
+
+    def register_submission(
+        self, submission: Submission, options: dict
+    ) -> Optional[dict]:
+        config = ObjectsAPIConfig.get_solo()
+        config.apply_defaults_to(options)
+
+        objects_client = config.objects_service.build_client()
+
+        created_object = objects_client.create(
+            "object",
+            {
+                "type": options["objecttype"],
+                "record": {
+                    "typeVersion": options["objecttype_version"],
+                    "data": {
+                        "data": submission.get_merged_data(),
```

Comment:
question: how do we deal with file uploads/attachments?

Reply:
I think we need to:

1) Allow a Documents API to be configured to upload attachment to; And relate the object to these documents (array of urls?)
2) Allow ZGW API's to be configured to either save the submission as PDF in the Documents API (as is now) **OR** save the submission in the Objects API. This also solves attachments because... ZGW.

Reply:
Uploading to Documents API & storing the resulting URLs in a field seems the simplest way to go for the time being, while still allowing us to make it more nice and expand later.

Reply:
Added this in https://github.com/open-formulieren/open-forms/pull/501/commits/1cbdb3b95445cf5cf32ecb116158b77d3dea3b61.

@sergei-maertens I tried some stuff locally and I can't seem to trigger the creation of a `SubmissionFileAttachment`, does this depend on the storage backend that is being used?

Reply:
The submission needs to be completed, and I believe it's part of the Celery task workflow to actually process the submissions.